pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -303,6 +312,118 @@ func (l *LibvirtDomainManager) setGuestTime(vmi *v1.VirtualMachineInstance) erro
 	return nil
 }
 
+// Set interfaces link down and up to renew DHCP leases
+func (l *LibvirtDomainManager) reconnectGuestNics(vmi *v1.VirtualMachineInstance) error {
+	logger := log.Log.Object(vmi)
+	domName := api.VMINamespaceKeyFunc(vmi)
+	dom, err := l.virConn.LookupDomainByName(domName)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error(failedReconnectGuestNic)
+		return err
+	}
+	defer dom.Free()
+	xmlstr, err := dom.GetXMLDesc(0)
+	if err != nil {
+		return err
+	}
+
+	var domain api.DomainSpec
+	err = xml.Unmarshal([]byte(xmlstr), &domain)
+	if err != nil {
+		return fmt.Errorf("parsing domain XML failed, err: %v", err)
+	}
+
+	// Look up all the interfaces and reconnect them
+	for _, iface := range domain.Devices.Interfaces {
+		specIface := getIfaceByName(vmi, iface.Alias.GetName())
+
+		// For masquerade and slirp we don't have to do anything
+		if specIface.Masquerade != nil || specIface.Slirp != nil {
+			continue
+		}
+
+		mac, changed, err := getIfaceMACAddress(iface)
+		if err != nil {
+			return fmt.Errorf("failed to get actual MAC address for network %s, err: %v", iface.Alias.GetName(), err)
+		}
+		if changed {
+			logger.Info("MAC address for " + iface.Alias.GetName() + " network is changed. Reattaching interfaces")
+			err = reattachIfaceWithMacAddress(dom, iface, mac)
```

Comment:
Does this work with multiple nics and multiple interfaces?  Or do we only need the first one we find?

Reply:
It works with all NICs, because you may want to use some traditional CNI-plagins like flannel. So IP address will be changed after the migration and should be renewed.